bib/internal/progress/progress.go

```diff
@@ -0,0 +1,296 @@
+package progress
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/cheggaaa/pb/v3"
+	"github.com/mattn/go-isatty"
+	"github.com/sirupsen/logrus"
+
+	"github.com/osbuild/images/pkg/osbuild"
+)
+
+var (
+	osStderr         io.Writer = os.Stderr
+	isattyIsTerminal           = isatty.IsTerminal
+)
+
+// ProgressBar is an interfacs for progress reporting when there is
+// an arbitrary amount of sub-progress information (like osbuild)
+type ProgressBar interface {
+	// SetProgress sets the progress details at the given "level".
+	// Levels should start with "0" and increase as the nesting
+	// gets deeper.
+	SetProgress(level int, msg string, done int, total int) error
+	// The high-level message that is displayed in a spinner
+	// (e.g. "Building image foo")
+	SetPulseMsg(fmt string, args ...interface{})
+	// A high level message with the last high level status
+	// (e.g. "Started downloading")
+	SetMessage(fmt string, args ...interface{})
+	Start() error
+	Stop() error
+}
+
+// New creates a new progressbar based on the requested type
+func New(typ string) (ProgressBar, error) {
+	switch typ {
+	case "", "plain":
+		return NewPlainProgressBar()
+	case "term":
+		return NewTermProgressBar()
+	case "debug":
+		return NewDebugProgressBar()
+	default:
+		return nil, fmt.Errorf("unknown progress type: %q", typ)
+	}
+}
+
+type termProgressBar struct {
+	spinnerPb   *pb.ProgressBar
+	msgPb       *pb.ProgressBar
+	subLevelPbs []*pb.ProgressBar
+
+	pool        *pb.Pool
+	poolStarted bool
+}
+
+// NewProgressBar creates a new default pb3 based progressbar suitable for
```

Comment:
NewProgressBar => NewTermProgressBar

or maybe even s/term/terminal? The short version is kinda confusing me, but it migth be a me-issue.